geomstats/geometry/special_euclidean.py

```diff
@@ -34,15 +36,101 @@
                         - 1. / 362880.]
 
 
-class SpecialEuclidean(LieGroup):
+class SpecialEuclidean(GeneralLinear, LieGroup):
+    """Class for special orthogonal groups."""
+
+    def __init__(self, n):
+        super(SpecialEuclidean, self).__init__(
```

Comment:
SpecialOrthogonal3 cannot be initilazied this way: because, then in the call to super in its init you get
`TypeError: super(type, obj): obj must be an instance or subtype of type`

Reply:
Is this because `SpecialOrthogonal3` is not added as a parent class of `SpecialOrthogonal`, therefore its constructor cannot be called?

There might be an error of class diagrams if we add it as a parent class of `SpecialOrthogonal`. In this case, we might want to create two privates classes `_SpecialOrthogonal3vector` and `_SpecialOrthogonaln` that will be then wrapped into the public class `SpecialOrthogonal`.

Reply:
And should the public class inherit from the private ones?